src/Integration/Helpers/ToolWIndowService.cs

```diff
@@ -51,6 +51,14 @@ public void EnsureToolWindowExists(Guid toolWindowId)
             GetOrCreateWindowFrame(toolWindowId);
         }
 
```

Comment:
Regarding testing, I am unsure how to mock a toolwindowpane. Is that something we do?

Reply:
@bigfluffycookie apologies, I missed this comment earlier.
Moq can mock classes as well as interfaces, as long as the class is not `sealed`. However, we'd generally try to mock an interface rather than a class - that way, we know we have complete control over everything the mock is returning.
If we are mocking a class, we can only override methods that the base class has marked as `virtual`. That might not give us the level of control we need for a test.

Fortunately, almost all of the public extension points in VS are interface-based, so we can almost always code against interfaces rather than classes. In this case, the `IToolWindowService` references `IVsWindowFrame` rather than `ToolWindowPane`, so we can mock that, which is what the existing `ToolWindowServiceTests` do (`ToolWindowPane` implements `IVsWindowFrame`).

In this case, it might be simpler just to create a real dummy class that implements `IVsWindowFrame` and has a `Guid` attribute on it e.g.

```
        [Guid("1D7B3C6A-BAB5-4A0D-8583-BCA4ECAEE649")]
        private class DummyToolWindowPaneWithGuid : IVsWindowFrame
        {
            public int Show() => throw new NotImplementedException();

            public int Hide() => throw new NotImplementedException();
            
            public int IsVisible() => throw new NotImplementedException();

            public int ShowNoActivate() => throw new NotImplementedException();
            "
            public int CloseFrame(uint grfSaveOptions) => throw new NotImplementedException();

            public int SetFramePos(VSSETFRAMEPOS dwSFP, ref Guid rguidRelativeTo, int x, int y, int cx, int cy) => throw new NotImplementedException();

            public int GetFramePos(VSSETFRAMEPOS[] pdwSFP, out Guid pguidRelativeTo, out int px, out int py, out int pcx, out int pcy) => throw new NotImplementedException();

            public int GetProperty(int propid, out object pvar) => throw new NotImplementedException();

            public int SetProperty(int propid, object var) => throw new NotImplementedException();

            public int GetGuidProperty(int propid, out Guid pguid) => throw new NotImplementedException();

            public int SetGuidProperty(int propid, ref Guid rguid) => throw new NotImplementedException();

            public int QueryViewInterface(ref Guid riid, out IntPtr ppv) => throw new NotImplementedException();

            public int IsOnScreen(out int pfOnScreen) => throw new NotImplementedException();
        }
```

It's might be possible to mock the `GuidAttribute` part using Moq, but sometimes it just easier to create a "real" class.